### PR TITLE
Adds new mod, Cataclysm: Craft Beers Ahead

### PR DESCRIPTION
**Pull request by charlesharris**
Cataclysm: Craft Beers Ahead is a new mod that adds a number of new beers (17 so far).  All of the new beers are associated with a brewpub or a microbrewery.  I've also added new generic beer styles that can be used to base new beers on.

> Full Moose Brewing - A microbrewery that generally produces a lot of strong beers, malty beers.  Fear the moose!

> North by Northeast - A brewpub with a balanced selection of beers that includes several IPAs and DIPAs.

> Dirndls and Dirndls - A brewpub putting out a focused selection of beers that are generally faithful representations of classic German beer styles. Also, dirndls.

I've tried to stick fairly close to the existing values for properties like "stim", "quench", "nutrition", etc., but there is natrually some variation from the generic beers.  

I've added the new beers to all of the `item_group`s that had beer(s) in them, but the spawn chance is all the same.  I'm certainly open to changing that if anyone thinks it's necessary.

If this isn't the right way to add a mod to the game and make it available for people, let me know and I'll follow the rules. :)

### CODE REVIEW

File: data/mods/C_CBA/modinfo.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "type": "MOD_INFO",
+    "mod_type": "SUPPLEMENTAL",
```
Comment (BrettDong) [inline data/mods/C_CBA/modinfo.json]: This line is unnecessary now iirc.

**Comment by BorkBorkGoesTheCode** [comment]
I have an early prototype for a basement brewery; would you be interested in including it if I upload it?

**Comment by Coolthulhu** [comment]
Technical details:
* In descriptions, periods should be followed by two spaces, not one
* Needs to be linted. Someone link the web linter - I don't have the link, I think
* Most of the beers share sets of stats: addiction type, materials, use action, phase, weight, price and so on. They should be defined using "copy-from". You can see examples of this in data/json/items/ammo/9mm.json. You define a base item which has all the stats all the beers share, then have all others copy from it and only overwrite what they need to overwrite.

**Comment by charlesharris** [comment]
@BorkBorkGoesTheCode  Sure!  I'm planning on doing a couple of brewpubs (generic and some that will have only their beer, e.g. North by Northeast), a microbrewery, a local homebrew store, a homebrewer's house with garage, and i think a multi-tile macrobrewery would be cool as well.

I also want to include some new recipes and the like, both for generic styles and clones of the named beers I'm introducing.  

That said, I'll probably keep it out of this initial PR if you don't mind, and add the buildings in later as a separate piece of work, hopefully sometime this weekend if I

**Comment by charlesharris** [comment]
@Coolthulhu Thanks for the feedback!  I'll take a look at those changes shortly.  

I actually was using `copy-from` originally, which was part of the reason I made the generic styles.  I had an issue somewhere in the JSON, though, and couldn't figure out what was missing.  I went and removed all of the `copy-from` usage and just hardcoded it all to remove anything I wasn't sure about.  By the time I realized what the JSON issue was, I had rebased and lost the original commit with the `copy-from` used.

Now I'm kicking myself for squashing too early. :)

**Comment by BorkBorkGoesTheCode** [comment]
@charlesharris : Okay, I will PR later.

>Needs to be linted. Someone link the web linter - I don't have the link, I think


http://dev.narc.ro/cataclysm/format.html

File: data/mods/C_CBA/modinfo.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "type": "MOD_INFO",
+    "ident": "c_cba",
+    "author": "Charles Harris",
```
Comment (keyspace) [inline data/mods/C_CBA/modinfo.json]: It is best to put your github name here, i.e. `charlesharris`.

**Comment by keyspace** [comment]
IMO it's best to lint with the provided Perl linter, since that's what the builbot uses anyway. The files-to-check list is `json_whitelist`.

Issue #20129 seems relevant.

File: data/mods/C_CBA/brewpubs/dirndls_and_dirndls/beers.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
+[
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_german_milkmaid",
+    "name" : "German Milkmaid",
+    "name_plural": "German Milkmaid",
+    "description" : "A classic Bavarian hefeweizen brewed by Dirndls and Dirndls brewpub. Delicious when served with a slice of orange.  Or not.  Also, dirndls.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/brewpubs/dirndls_and_dirndls/beers.json]: Two spaces after period, please.

File: data/mods/C_CBA/brewpubs/dirndls_and_dirndls/beers.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_ups_pilsner",
+    "name" : "UPS Pilsner",
+    "name_plural": "UPS Pilsner",
+    "description" : "A classic German pilsner brewed by Dirndls and Dirndls brewpub.  Clean, crisp, and made to be consumed by the liter. Helps you power through the day and late into the night. Probably too late into the night... Also, dirndls.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/brewpubs/dirndls_and_dirndls/beers.json]: And here.

File: data/mods/C_CBA/brewpubs/north_by_northeast/beers.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
+[
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_npc_double_esb",
+    "name" : "NPC Double ESB",
+    "name_plural": "NPC Double ESB",
+    "description" : "A Double Extra Special Bitter brewed by North by Northeast brewpub.  A terribly seductive, deceptive brew. It is complex and it seems to speak to you on an almost intelligent, rational level. Then you wake up on fire.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/brewpubs/north_by_northeast/beers.json]: And here.

File: data/mods/C_CBA/brewpubs/north_by_northeast/beers.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_extra_dimensional_hopwizard",
+    "name" : "Extra-dimensional Hopwizard",
+    "name_plural": "Extra-dimensional Hopwizard",
+    "description" : "A Septuple IPA brewed by North by Northeast brewpub. Yer a wizard, 'arry!  A freaking hop wizard!",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/brewpubs/north_by_northeast/beers.json]: And here.

File: data/mods/C_CBA/brewpubs/north_by_northeast/beers.json [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_ahopalypse_now",
+    "name" : "Ahopalypse Now",
+    "name_plural": "Ahopalypse Now",
+    "description" : "A Double IPA brewed by North by Northeast brewpub. Very smooth for a Double IPA. You've got a fever... and the only cure... is more hops.  May incite spontaneous dancing.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/brewpubs/north_by_northeast/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 8]
```diff
@@ -0,0 +1,152 @@
+[
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_moosecuddler",
+    "name" : "Moosecuddler",
+    "name_plural": "Moosecuddler",
+    "description" : "A barrel-aged Russian imperial stout brewed by Full Moose Brewing. Life's a moose, then you die!",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "weight" : 249,
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_el_moosenator",
+    "name" : "El Moosenator",
+    "name_plural": "El Moosenator",
+    "description" : "A German bock brewed by Full Moose Brewing. Strong, malty, and kicks ya right in the teeth.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_hair_of_the_moose",
+    "name" : "Hair of the Moose",
+    "name_plural": "Hair of the Moose",
+    "description" : "A dark session ale brewed by Full Moose Brewing. Had a rough night? Full of regrets? Here's a dark, smooth session ale that's easy to drink to help you feel better. Just close your eyes and knock back a few of these. What you can't see, can't hurt you, right?",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_ninja_moose",
+    "name" : "Ninja Moose",
+    "name_plural": "Ninja Moose",
+    "description" : "A Belgian dubbel brewed by Full Moose Brewing. Dark, malty and strong. This one'll sneak right up on you like a ninja. Or a moose. Or a ninja moose.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_moose_moose_moose",
+    "name" : "M3",
+    "name_plural": "M3",
+    "description" : "A Belgian tripel brewed by Full Moose Brewing. Good things come in threes. Or was it bad things? Doesn't really matter, just have another one of these and somehow everything just seems totally okay.",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

File: data/mods/C_CBA/microbreweries/full_moose/beers.json [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+    "volume" : 1,
+    "phase" : "liquid",
+    "flags" : ["EATEN_COLD"]
+  },
+  {
+    "type" : "COMESTIBLE",
+    "id" : "beer_moose_moose_moose_x3",
+    "name" : "M9",
+    "name_plural": "M9",
+    "description" : "The barrel-aged version of M3. Stronger, faster, smarter, better, another!",
```
Comment (Night-Pryanik) [inline data/mods/C_CBA/microbreweries/full_moose/beers.json]: And here.

**Comment by BrettDong** [comment]
In `modinfo.json`, you need to write in this way:
```
"authors": [ "CharlesHarris" ],
"dependencies": [ "dda " ],
```
`authors` has a s.

**Comment by BorkBorkGoesTheCode** [comment]
@charlesharris As BrettDong said. The format would resemble this:

```
[
  {
    "type": "MOD_INFO",
    "ident": "c_cba",
    "authors": [ "Charles Harris" ],
    "name": "Cataclysm: Craft Beers Ahead",
    "description": "Adds new content that includes a selection of seventeen new craft beers.  There are also new generic beer styles that can be used as reference for new beers.  Additional planned content which has not yet been implemented includes new locations (brewpub, microbrewery, a local homebrew store, a homebrewer's house, etc.), books, and recipes.",
    "category": "items",
    "dependencies": [ "dda" ]
  }
]

```
The `"dependencies":` part was added recently as part of the push to allow total conversion mods. It links back to the default Cataclysm: DDA modpack. `"authors":` is the new way of allowing arrays.

**Comment by Coolthulhu** [comment]
A long list of unfixed issues here. Looks abandoned.
If it isn't, fix the issues and re-PR.

**Pull request closed without merging**